frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,353 @@
+  id: string;
+  name: string;
+};*/
+
+export type PvcState = {
+  resourceObj: K8sResourceKind;
+  inProgress: boolean;
+  pvcName: string;
+  accessMode: string;
+  requestSizeValue: string;
+  requestSizeUnit: string;
+  defaultRequestSizeValue: string;
+  defaultRequestSizeUnit: string;
+  error?: any;
+  disableForm: boolean;
+  radios: any;
+  dropdownUnits: any;
+  showKeyValue: boolean;
+  nameValuePairs: any;
+};
+
+export type PvcProps = {
+  kindObj: K8sKind;
+  isCreate: boolean;
+  saveButtonText?: string;
+  namespace: string;
+};
+
+class CreatePvcForm extends React.Component<PvcProps, PvcState> {
+  constructor(props) {
+    super(props);
+    const resourceObj = {
+      apiVersion: 'v1',
+      kind: 'PersistentVolumeClaim',
+      metadata: {
+        name: '',
+        namespace: '',
+      },
+      spec: {
+        accessModes: {},
+        resources: {
+          requests: {
+            storage: '',
+          },
+        },
+        selector: {
+          matchLabels: {},
+        },
+      },
+    };
+
+    this.state = {
+      resourceObj: resourceObj,
+      inProgress: false,
+      pvcName: '',
+      accessMode: '',
+      requestSizeValue: '',
+      requestSizeUnit: '',
+      defaultRequestSizeValue: '1',
+      defaultRequestSizeUnit: 'Gi',
+      disableForm: false,
+      showKeyValue: false,
+      nameValuePairs: [['', '']],
+      radios: [
+        {
+          value: 'ReadWriteOnce',
+          title: <span className="form-right-padding">Single User (RWO)</span>,
+        },
+        {
+          value: 'ReadWriteMany',
+          title: <span className="form-right-padding">Shared Access (RWX)</span>,
+        },
+        {
+          value: 'ReadOnlyMany',
+          title: <span className="form-right-padding">Read Only (ROX)</span>,
+        },
+      ],
+      dropdownUnits: {
+        ['Mi']: 'Mi',
+        ['Gi']: 'Gi',
+        ['Ti']: 'Ti',
+      },
+    };
+  }
+  componentDidMount() {
+    this.setState({
+      accessMode: 'ReadWriteOnce',
+      requestSizeValue: this.state.defaultRequestSizeValue,
+      requestSizeUnit: this.state.defaultRequestSizeUnit,
+    });
+  }
+  onError = err => {
+    this.setState({
+      error: err,
+      inProgress: false,
+    });
+  };
+
+  handleChange = event => {
+    // this handles pvcName, accessMode, size
+    const name = event.target.name;
+    const value = event.target.value;
+    this.setState({ [name]: value } as Pick<PvcState, keyof PvcState>);
+  };
+
+  handleNameValuePairs = event => {
+    this.setState({ nameValuePairs: event.nameValuePairs });
+  };
+
+  onShowLabelSelectors = () => {
+    this.setState({ showKeyValue: !this.state.showKeyValue });
+  };
+
+  onFormDisable = disable => {
+    this.setState({
+      disableForm: disable,
+    });
+  };
+
+  validateForm = () => {
+    const { pvcName, accessMode, requestSizeUnit, requestSizeValue } = this.state;
+    // Check that a valid PVC has been choosen
+    if (!pvcName) {
```

Comment:
Since the input fild for the pvc name is required, I dont think this validation is needed

Reply:
pvcName does not have a default so I feel it should have a check even though the form prevents it.   For example, what if we had some unit test that was just calling the save?